cowprotocol/cowswap
# [ETH-FLOW V2] - Slippage updater

# Summary

Replaces #1185 now that the rework is merged

1. Creates an updater to check eth flow state and set slippage to 2% if eth-flow and back to previous erc20<>erc20 slippage amt when changed
2. Edits `TransactionSettingsMod` to block slippage changes if `isEthFlow`

## Screenshots
NATIVE SWAP DETECTED:
(slippage showing modified and @ 2%)
<img width="481" alt="image" src="https://user-images.githubusercontent.com/21335563/195191783-ac06bc4d-cd43-4536-a977-1de2752e7ac6.png">
(transaction settings showing disabled input/auto select option)
<img width="354" alt="image" src="https://user-images.githubusercontent.com/21335563/195191912-5f4a5d10-b5e9-48bf-bb61-23c775567ed6.png">

# To Test
1. Select ETH <> ERC20
2. Slippage shows "modified" tag and is 2%
3. open slippage settings
4. slippage settings should not be modifiable
5. Select ERC20<>ERC20
6. slippage should go back to what it was previous to eth flow
7. slippage settings should be editable

Note: keep in mind the backend isn't ready yet, so clicking swap will ALWAYS show u the wrap flow and slippage will be reset back to the WETH selected value.
